### ISSUE TITLE
a question about pipe unpack_json and filter

### ISSUE THREAD

**Issue by bearslyricattack** [VictoriaMetrics/VictoriaMetrics#7915]
### Is your question request related to a specific component?

about Victorialogs

### Describe the question in detail

Is it impossible to connect the filter pipeline after unpack_json pipeline to implement filtering？

### Troubleshooting docs

- [X] General - https://docs.victoriametrics.com/troubleshooting/
- [ ] vmagent - https://docs.victoriametrics.com/vmagent/#troubleshooting
- [ ] vmalert - https://docs.victoriametrics.com/vmalert/#troubleshooting

**Comment by tiny-pangolin** [comment]
Yes, this is possible if you have a json log like the one below you can filter based on a field created by extract_json by adding `| <filter>` to your query so the query `tags.appname:traefik | extract_json` would turn into `tags.appname:traefik | extract_json | ClientHost: 10.0.0.7`. It is more performant to have your log collector parse the data and store it as [other fields](https://docs.victoriametrics.com/victorialogs/keyconcepts/#other-fields), but this approach will work. 

```json
{"ClientAddr":"10.0.0.7:7105","ClientHost":"10.0.0.7","ClientPort":"7105","ClientUsername":"-","DownstreamContentSize":0,"DownstreamStatus":401,"Duration":915469,"OriginContentSize":0,"OriginDuration":701655,"OriginStatus":401,"Overhead":213814,"RequestAddr":"grafana.local:443","RequestContentSize":0,"RequestCount":121815573,"RequestHost":"grafana.local","RequestMethod":"GET","RequestPath":"/metrics","RequestPort":"443","RequestProtocol":"HTTP/1.1","RequestScheme":"https","RetryAttempts":0,"RouterName":"grafana@file","ServiceAddr":"grafana:3000","ServiceName":"grafana@file","ServiceURL":"http://grafana:3000","SpanId":"0000000000000000","StartLocal":"2024-12-23T16:45:10.337597217Z","StartUTC":"2024-12-23T16:45:10.337597217Z","TLSCipher":"TLS_AES_128_GCM_SHA256","TLSVersion":"1.3","TraceId":"00000000000000000000000000000000","entryPointName":"websecure","level":"info","msg":"","time":"2024-12-23T16:45:10Z"}
```

**Comment by bearslyricattack** [comment]
Thank you, but I couldn't find any knowledge about the extract_comson pipeline in the document. Also, I would like to know if all filters treat log entries as strings. If you want to turn them into JSON, you will need to use the JSON pipeline. After using this pipeline, you cannot filter based on the fields in the results of this pipeline?

**Comment by tiny-pangolin** [comment]
> but I couldn't find any knowledge about the extract_comson
I made a mistake, the correct pipe would be `unpack_json` not `extract_json` and [here are the docs for the unpack_json pipe](https://docs.victoriametrics.com/victorialogs/logsql/#unpack_json-pipe)

>  I would like to know if all filters treat log entries as strings
I believe each message is treated as a string, but certain pipes and filters will attempt to treat parts of the message like a number like [stats pipes](https://docs.victoriametrics.com/victorialogs/logsql/#stats-pipe) and the [range pipe](https://docs.victoriametrics.com/victorialogs/logsql/#range-filter)

> After using this pipeline, you cannot filter based on the fields in the results of this pipeline?
you can add additional filters either by adding an `AND` like `tags.appname:traefik | extract_json | ClientHost: 10.0.0.7 AND ClientPort:7105` or by adding an additional pipe like `tags.appname:traefik | extract_json | ClientHost: 10.0.0.7 | ClientPort:7105`
